src/main/resources/static/javascript/mainApp/general.controller.js

```diff
@@ -1476,11 +1474,51 @@
             });
         };
 
-        function handleGroupingReset() {
-            $scope.getGroupingInformation();
-            $scope.loading = false;
-            $scope.createResetNotifModal($scope.selectedGrouping.name);
-        }
+
+        $scope.displayUnsuccessfulGroupResetModal = function (message) {
+            $scope.launchDynamicModal(Message.ResetGroupError.ResetGroupErrorMessages.TITLE, message);
+        };
+        /**
+         * Helper-Guard: handleGroupingReset:
+         * If there are error results, render an error modal and return true, otherwise return false.
+         */
+        $scope.handleGroupingReset = function (groupingPath, resetIncludeResult, resetExcludeResult) {
+            let hasErrors = function (result) {
+                return (typeof result !== "undefined" && typeof result.status !== "undefined");
+            };
+            let checkResetResult = (groupingPath, resetResult) => {
+                return (typeof resetResult !== "undefined") &&
+                    (resetResult.resultCode === "SUCCESS" && resetResult.groupPath.includes(groupingPath));
+            };
+            let results = {
+                "includeFailure": hasErrors(resetIncludeResult),
+                "excludeFailure": hasErrors(resetExcludeResult),
+                "includeSuccess": checkResetResult(groupingPath, resetIncludeResult),
+                "excludeSuccess": checkResetResult(groupingPath, resetExcludeResult)
+            };
+            if (results.includeFailure || results.excludeFailure) {
+                for (let i = 0; i < Message.ResetGroupError.ResetGroupErrorMessageMap.length; i++) {
+                    let object = Message.ResetGroupError.ResetGroupErrorMessageMap[parseInt(i, 10)];
+                    if (Utility.compareObjects(object, results)) {
+                        let message = Message.ResetGroupError.ResetGroupErrorMessages.Body[parseInt(i, 10)];
```

Comment:
You can use a for of loop here to avoid using parseInt(i, 10) to fix the Codacy issue.